src/helpers/sorobanUtils.ts

```diff
@@ -0,0 +1,174 @@
+      durability: getXdrDurability(durability),
+    }),
+  );
+};
+
+export const getSorobanDataResult = ({
+  contractDataXDR,
+  operationType,
+  fee,
+}: {
+  contractDataXDR: xdr.LedgerKey;
+  operationType: SorobanOpType;
+  fee: string;
+}) => {
+  switch (operationType) {
+    case "extend_footprint_ttl":
+      return buildSorobanData({
+        readOnlyXdrLedgerKey: [contractDataXDR],
+        resourceFee: fee,
+      });
+    // case "restore_footprint":
+    // return buildSorobanData({
+    //   readWriteXdrLedgerKey: [contractDataXDR],
+    //   resourceFee: operation.params.resource_fee,
+    // });
+    default:
+      return undefined;
+  }
+};
+
+export const buildSorobanTx = ({
+  sorobanData,
+  params,
+  sorobanOp,
+  networkPassphrase,
+}: {
+  sorobanData: xdr.SorobanTransactionData;
+  params: TransactionBuildParams;
+  sorobanOp: TxnOperation;
+  networkPassphrase: string;
+}) => {
+  // decrement seq number by 1 because TransactionBuilder.build()
+  // will increment the seq number by 1 automatically
+  const txSeq = (BigInt(params.seq_num) - BigInt(1)).toString();
+  const account = new Account(params.source_account, txSeq);
+
+  // https://developers.stellar.org/docs/learn/fundamentals/fees-resource-limits-metering
+  const totalTxFee = BigInt(params.fee) + BigInt(sorobanOp.params.resource_fee);
+
+  const getMemoValue = (memoType: string, memoValue: string) => {
+    switch (memoType) {
+      case "text":
+        return Memo.text(memoValue);
+      case "id":
+        return Memo.id(memoValue);
+      case "hash":
+        return Memo.hash(memoValue);
+      case "return":
+        return Memo.return(memoValue);
+      default:
+        return Memo.none();
+    }
+  };
+
+  const getTimeboundsValue = (timebounds: {
+    min_time: string;
+    max_time: string;
+  }) => {
+    return {
+      minTime: timebounds.min_time,
+      maxTime: timebounds.max_time,
+    };
+  };
+
+  const transaction = new TransactionBuilder(account, {
+    fee: totalTxFee.toString(),
+    timebounds: getTimeboundsValue(params.cond.time),
+  });
+
+  if (Object.keys(params.memo).length > 0) {
+    const [type, val] = Object.entries(params.memo)[0];
+    transaction.addMemo(getMemoValue(type, val));
+  }
+
+  return transaction
+    .setNetworkPassphrase(networkPassphrase)
+    .setSorobanData(sorobanData)
+    .addOperation(
+      Operation.extendFootprintTtl({
+        extendTo: Number(sorobanOp.params.extend_ttl_to),
+        source: sorobanOp.source_account,
+      }),
+    )
+    .build();
+};
+
+const buildSorobanData = ({
+  readOnlyXdrLedgerKey = [],
+  readWriteXdrLedgerKey = [],
+  resourceFee,
+  //   instructionsm
+  //   ReadableByteStreamController,
+}: {
+  readOnlyXdrLedgerKey?: xdr.LedgerKey[];
+  readWriteXdrLedgerKey?: xdr.LedgerKey[];
+  resourceFee: string;
+}) => {
+  // one of the two must be provided
+  if (!readOnlyXdrLedgerKey && !readWriteXdrLedgerKey) {
+    return;
+  }
```

Comment:
Should we check for length here since we're setting default values as empty arrays?